src/fsharp/nameres.fs

```diff
@@ -2452,8 +2452,17 @@ let ResolveRecordOrClassFieldsOfType (ncenv: NameResolver) m ad typ statics =
     |> List.filter (fun rfref -> rfref.IsStatic = statics  &&  IsFieldInfoAccessible ad rfref)
     |> List.map Item.RecdField
 
+[<RequireQualifiedAccess>]
+type ResolveCompletionTargets =
+    | All of (MethInfo -> TType -> bool)
+    | SettablePropertiesAndFields
+    member this.ResolveAll = 
+        match this with
+        | All _ -> true
+        | _ -> false
```

Comment:
side question: I always wondered why RequireQualifiedAccess doesn't work on members of the union type. Is this a good thing?